### DISCUSSION TITLE
Member role list empty

### DISCUSSION THREAD

**Discussion by TryhardOne** [Discord4J/Discord4J#926] [Q&A] [answered]
Hello, I was trying to implement a self role feature for my bot.

For this I retrieve the role ids of a member multiple times and load them into an arraylist, i do this via;

	private void reInitRoles() {
		userRoleIDs.clear();
		System.out.println(userID);
		Set<Snowflake> set = Core.client.getMemberById(Snowflake.of(guildID), Snowflake.of(userID)).block()
				.getRoleIds();
		Iterator<Snowflake> it = set.iterator();
		while (it.hasNext()) {
			userRoleIDs.add(it.next().asString());
		}
		System.out.println("ARRAYLIST ROLES: " + userRoleIDs);
	}

The problem is that if I retrieve the users roles after adding a role to the user, or removing one, the list only contains the roles that the user had at the moment the bot connected to the server.
So if the user had no roles, the list is empty, if I add a role and retrieve the role ids, the list is still empty, unless I restart the bot.

**Comment by darichey** [comment, marked as answer]
This is related to #896. You'll need to enable privileged intents for now.

**Reply by TryhardOne** [reply]
this absolutely fixes it, didnt expect that kind of problem ;D thanks
